src/box/session.cc

```diff
@@ -364,8 +409,11 @@ session_init(void)
 {
 	session_registry = mh_i64ptr_new();
 	mempool_create(&session_pool, &cord()->slabc, sizeof(struct session));
+	fiber_cond_create(&on_shutdown_list_empty);
 	credentials_create(&admin_credentials, admin_user);
 	session_settings_init();
+	if (box_on_shutdown(NULL, session_on_shutdown_f, NULL) != 0)
+		unreachable();
```

Comment:
may be `diag_log` here before `unreachable`?

Reply:
`box_on_shutdown` doesn't set diag - it logs an error instead - so `diag_log` wouldn't work here. Anyway, I replaced `unreachable` with `panic`.